Geometry/CaloGeometry/src/FlatTrd.cc

```diff
@@ -96,6 +96,18 @@ GlobalPoint FlatTrd::getPosition(const Pt3D& local) const {
   return GlobalPoint(glb.x(), glb.y(), glb.z());
 }
 
+float FlatTrd::etaSpan() const {
+  float eta1 = -std::log(std::tan(0.5 * std::atan((m_global.perp() + param()[k_dY1]) / std::abs(m_global.z()))));
+  float eta2 = -std::log(std::tan(0.5 * std::atan((m_global.perp() - param()[k_dY1]) / std::abs(m_global.z()))));
+  float dEta = std::abs(eta1 - eta2);
+  return dEta;
+}
+
+float FlatTrd::phiSpan() const {
+  float dPhi = 2.0 * std::atan(0.5 * (param()[k_dX1] + param()[k_dX2]) / m_global.perp());
```

Comment:
For all four versions of `etaSpan` and `phiSpan`, please prevent segmentation faults and nonsense values. Please check:
```
assert(param() != nullptr);
```
and check that `m_global.z()` and `m_global.perp()` are not zero before dividing by them.

Reply:
Done